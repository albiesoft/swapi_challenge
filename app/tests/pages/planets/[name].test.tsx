import { shallow, ShallowWrapper } from 'enzyme';
import preloadAll from 'jest-next-dynamic';
import PlanetsPage from '../../../pages/planets/[name]';

describe('<PlanetsPage />', () => {
  beforeAll(async () => {
    await preloadAll();
  });

  test('It renders correctly', () => {
    const data = {
        Name: 'PlanetName',
        Climate: 'Hot',
        Population: 10000,
        Films: [
          {
            Title: 'FilmTitle',
            Director: 'FilmDirector',
            ReleaseDate: '2020-03-08'
          }
        ]
    }
    let target: ShallowWrapper = shallow(<PlanetsPage planet={data} />);
    expect(target).toMatchSnapshot();
  });
});
