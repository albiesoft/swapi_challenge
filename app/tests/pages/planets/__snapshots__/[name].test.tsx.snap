// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlanetsPage /> It renders correctly 1`] = `
<MainLayout>
  <Head>
    <title>
      Planet 
      PlanetName
    </title>
  </Head>
  <section
    className="planet-info"
  >
    <h1>
      <ForwardRef(RocketTwoTone) />
       
      PlanetName
    </h1>
    <p>
      <ForwardRef(SmileTwoTone) />
       Population: 
      10000
    </p>
    <p>
      <ForwardRef(FireTwoTone) />
       Climate: 
      Hot
    </p>
  </section>
  <Row
    className="planet-films"
    gutter={16}
  >
    <Col
      key="FilmTitle"
      lg={8}
      md={12}
      sm={24}
      xs={24}
    >
      <Card
        style={
          Object {
            "marginBottom": "16px",
          }
        }
        title="FilmTitle"
      >
        <p>
          Realese Date: 
          2020-03-08
        </p>
        <p>
          Director: 
          FilmDirector
        </p>
      </Card>
    </Col>
  </Row>
</MainLayout>
`;
